pages/addEventListener_hook.coffee

```diff
@@ -0,0 +1,44 @@
+_addEventListener = Element::addEventListener
+
+loaded = false
+elementsToRegister = []
+registrationElement = null
+
+EventTarget::addEventListener = (type, listener, useCapture) ->
+  eventTarget = if this in [document, window] then document.documentElement else this
+  if type == "click" and eventTarget instanceof Element
+    setTimeout (-> registerElementWithContentScripts eventTarget, "onclick"), 0
+  _addEventListener.apply this, arguments
+
+onLoaded = ->
+  loaded = true
+
+  # Create an element detatched from the DOM, register it with the content scripts.
+  registrationElement = document.createElementNS "http://www.w3.org/1999/xhtml", "div"
+  registrationEvent = new CustomEvent "VimiumRegistrationElementEvent"
+  document.documentElement.appendChild registrationElement
+  registrationElement.dispatchEvent registrationEvent
```

Comment:
The only way to pass an element between Chrome's isolated worlds seems to be as an event's target, so we have to dispatch on the element where `addEventListener` was called.

The basic logic of this code is:
* if the element is in the DOM, dispatch an event on it and catch it in the content script as it bubbles to `document`, or
* if the element isn't in the DOM, we need to add it to somewhere that has an event listener that the event will bubble up to. We don't want to add it to `document` since we don't want to render elements prematurely or trigger lots of `MutationObserver` calls. Thus, the solution I came up with is:
  * communicate `registrationElement` to the content script (which the line you commented on is doing), so it can add a event listener
  * add elements not in the DOM to `registrationElement` so that the event listener catches the events, then remove them again so the page can still add them to the DOM at will.